CIP-0030/README.md

```diff
@@ -76,10 +97,17 @@ type Paginate = {|
 ```
 Used to specify optional pagination for some API calls. Limits results to {limit} each page, and uses a 0-indexing {page} to refer to which of those pages of {limit} items each. dApps should be aware that if a wallet is modified between paginated calls that this will change the pagination, e.g. some results skipped or showing up multiple times but otherwise the wallet must respect the pagination order.
 
+#### Extension
+
+An extension is an object with a single field `"cip"` that describe a CIP number extending the API (as a plain integer, without padding). For example:
```

Comment:
And you don't have to if you do not want. You can always ignore it and do with whatever the wallet provides you. Yet, the mechanism exists for those who seek this granularity.

Reply:
An alternative would be to require each CIP to unambiguously define a slug that describes the API like "notifications v1" but this is even less discoverable or consistent.  In practice the cardano community has rallied around referring to things by CIP number anyway ("is the wallet CIP-30 compliant?", "Is this a CIP-68 compatible NFT", etc.)

Even if you're unaware of the details or history of the CIP, you're aware of the name, which gives you a shared frame for googling, discussing, finding libraries for, etc.

Reply:
You'd still have to look up the cip(s) for what's supported in which versions. Don't see a difference in terms of dApp developer effort.

We can (and will) build libraries on top of dApp connector cip(s), which can abstract away the cip(s) and have opinionated APIs. Having said that, I'm not sure we should use the current revision cip-0030 as an immutable base. For example, all wallets have to implement `getCollateral` to be compliant, but it should be deprecated at some point, as using 'collateral return' is a superior way of handling the collateral (any utxo can be used as a collateral).

@KtorZ how about drafting a new CIP that would define the dapp connector without **any** wallet functionality, but with extensions, where cip-0030 wallet API is one of the extensions?

Reply:
@mkazlauskas that would have been the ideal approach. But given that CIP-0030 is already somewhat implemented by everyone already. I am arguably just trying to make it slightly easier to manage all conversations down-the-line w.r.t adding stuff to CIP-0030. From a CIP perspective, it's been a bit hard to manage evolutions of CIP-0030 precisely because adding anything to it pushes work onto everyone and it's hard to find consensus. 

Having said that, I wouldn't be against _trimming down_ the current CIP-0030 to a bare minimum, and move core functionality into separate CIPs. Yet I have not really the bandwidth to write and champion those other CIPs (plural because I think the current API should rather be expressed as separate points).

Reply:
CIP-30 certainly isn't an immutable base; a future CIP could replace/modify it, and then dApps would signify they're ready to upgrade by passing in that CIP number, and wallets would gradually support it.

Eventually we may get to a state where in practice nobody passes an empty array, and no wallets supports the very old CIP-30 and fail if you do pass an empty array.

It "feels" unclean because you'll have some APIs hanging around for a long time, but in practice that's what backwards compatibility, graceful evolution, and long term support *means*.

Another way to think about it:
 
CIP-30 is just another extension you're enabling, but it's assumed to be passed by default for backwards compatibility.

Reply:
Yup

Reply:
> how about drafting a new CIP that would define the dapp connector without any wallet functionality, but with extensions, where cip-0030 wallet API is one of the extensions?

I am keen to take a swing at this and see what people think 😎

Reply:
@Ryun1 @KtorZ I think that will save _so_ much grief in the long run... imagine if, in its own early days, developers had made an inventory of the few protocols which would "always" be in the Internet: and how unworkable that would seem as a baseline today 🧐

Reply:
Resolved by https://github.com/cardano-foundation/CIPs/pull/462 (@thaddeusdiamond)?  At least with respect to the feature baseline (leaving aside the nomenclature issues)...?

Reply:
> Resolved by #462 (@thaddeusdiamond)? At least with respect to the feature baseline (leaving aside the nomenclature issues)...?

It is resolved in terms of this issue, but I'd actually rather that linked specification use the slugs as someone mentioned below `collateral-experimental`.  Naming is hard enough, linking them to numbers adds yet another layer of obscurity.

It makes me think of the old CS saying: "There are only two hard problems in computer-science: floating points, naming, and off-by-one errors".

Reply:
@thaddeusdiamond then I'm resolving this thread & can you or @Quantumplation please post your suggestion about the naming convention in #462? (I don't see it there) cc: @Ryun1

Reply:
@thaddeusdiamond then I'm resolving this thread & can you or @Quantumplation please post your suggestion about the naming convention in #462? (I don't see it there) cc: @Ryun1